### DISCUSSION TITLE
Time Code Jam

### DISCUSSION THREAD

**Discussion by chaudhuri-ab** [gopro/labs#445] [Q&A] [answered]
I have a quick question regarding jamming a Hero 9 via 3.5mm input.

When doing this and the camera is put to sleep or turned off will the internal clock stay in sync when the device is powered up again or do I have to jam the go pro again if it powers off?

Thanks

**Comment by dnewman-gpsw** [comment, marked as answer]
After the camera sleeps (shuts-down) the timecode is only within a +/- one second of accuracy.  For reliable timecode precision, the camera need to be remain on after the jam-sync.

**Reply by chaudhuri-ab** [reply]
@dnewman-gpsw - I perhaps should post this in features requested, but a thought I had is can the QR control app be updated to generate a Time Code Signal output on an iPhone/Android such that someone could Jam a Go Pro via the QR code and perhaps another device via 3.5mm audio out?

If there is an app that does this already, please do share, but the only app I found that used the Go Pro QR was Tentacle only if you have one of their devices already.

The use case I have for this here is I have a Zoom M4 and would like to use it for audio with the Go Pro. For things like motorcycle rides or landscape shots with sound (e.g. Ocean) it is nice to have good enough time code syncing that is better than a hand clap, but does not need to be as precise as extended dialog. The M4 and similar recorders can take in a TC signal via 3.5 mm and  it would be really useful if the QR Control App could output via an iPhone/Android audio out a TC signal alongside the QR code so that you can Jam other devices alongside the Go Pro without having to mess with the 3.5 mm input on the Go Pro (i.e. the QR Jamming on the Go Pro is really well done and preferable to use).


Thanks

**Reply by dnewman-gpsw** [reply]
I have considered doing that (I'm the QRControl developer), the idea is in the back log.  It should be fairly easy to create a standalone app that generates QR Codes time and LTC over an audio jack.  Any developers out there want to do it?

**Reply by chaudhuri-ab** [reply]
> Any developers out there want to do it?

A couple question regarding this:

- Does the jam sync QR code basically contain just precision time that the Go Pro pics up (e.g. oT230715164940.07)?

- Is precision time setting the same mechanism that the OS uses to keep time code in sync via 3.5 mm on the go pro if an input is constantly provided to the Go Pro?

- It seems that to jam sync the Go Pro and let us say other devices via LTC via 3.5mm, it would be better to disable the GPS time sync feature on the Go Pro right (Note - I am not sure how the Go Pro OS uses the GPS UTC time and what clocks it updates. It seems like if you jam sync in the manner outlined, disconnect all devices from generators, then you would want to avoid the Go Pro getting a GPS time adjustment post syncing. However, this all depends on what precisely the OS does with the GPS UTC time and what is or is not updated)? 

- Does the Go Pro normal firmware update time via GPS automatically or is that something that only is available in Go Pro labs firmware?
